app/controllers/application_controller.rb

```diff
@@ -4,8 +4,18 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :set_app_info_headers
 
+  private
+
   def set_app_info_headers
     headers["X-GitHub-Repository"] = "https://github.com/department-of-veterans-affairs/vets-api"
     headers["X-Git-SHA"] = AppInfo::GIT_REVISION
   end
+
+  def require_login
+    unless session[:user]
+      flash[:after_login_controller] = request.parameters["controller"]
```

Comment:
The flash uses ActionDispatch::Session::CacheStore, which is a cookie. I think this solution is just-for-now, but perhaps it would be good to be a TODO: in here to suggest that this will do something with JSON instead. We probably don't want to use flash or session cookies in this app.